src/numerics/CVodesIntegrator.cpp

```diff
@@ -513,18 +546,27 @@ void CVodesIntegrator::integrate(double tout)
                 "Components with largest weighted error estimates:\n{}",
                 flag, m_error_message, f_errs, getErrorInfo(10));
         }
+        if (flag == CV_ROOT_RETURN) {
+            // Stop early at root (e.g., advance limit reached)
+            root_triggered = true;
+            break;
+        }
```

Comment:
I think you could slightly simplify the logic here by just setting `tout = m_tInteg` and eliminating the `root_triggered` flag.